schematics/ng-add/setup-project.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  chain,
+  Rule,
+  SchematicsException,
+  Tree,
+} from '@angular-devkit/schematics';
+import {addImportToModule} from '@schematics/angular/utility/ast-utils';
+import {InsertChange} from '@schematics/angular/utility/change';
+import {getWorkspace} from '@schematics/angular/utility/config';
+import {getAppModulePath} from '@schematics/angular/utility/ng-ast-utils';
+import {getProject} from '@schematics/angular/utility/project';
+import * as path from 'path';
+import * as ts from 'typescript';
```

Comment:
Recently fixed issue in another library, which was caused by the similar code. `@schematics/angular` bundles its own version of TS and as a result AST is created by the workspace version of TS (`ts.createSourceFile()` call), while it is modified by the version bundled by `@schematics/angular`. If these versions are different it crashes. The reproduction was to `ng new my-proj` with latest CLI, change workspace TS to 3.7.5 and `ng add <lib>`.

See https://github.com/FortAwesome/angular-fontawesome/pull/238 for more details.

Reply:
Thank you, will check this